docs/scheduler.md

```diff
@@ -0,0 +1,152 @@
+
+### Distributed jobs
+
+Distributed jobs are represented as [PodGroups](https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/kep/42-podgroup-coscheduling/README.md), i.e., groups of cooperating pods, and the pods that make up a `PodGroup` are scheduled jointly using a gang-scheduling algorithm. Similar to to [kube-batch](https://github.com/kubernetes-sigs/kube-batch) and [Volcano](https://volcano.sh/en/docs/podgroup/), the gang-scheduling algorithm is implemented as a [custom Kubernetes scheduler](https://kubernetes.io/docs/tasks/extend-kubernetes/configure-multiple-schedulers/).
+
+### Elasticity and dynamic resource creation
+
+Armada supports workflows that require a dynamically changing amount of resources (e.g., [Apache Spark](https://spark.apache.org/) clusters and AirFlow jobs) using the following two approaches:
+
+- The workflow is submitted to a higher-level scheduler (e.g., Apache AirFlow) that dynamically submits jobs to Armada, monitors their status, and makes decisions on which jobs to submit next.
+- An "operator" job is submitted to Armada (e.g., corresponding to a Spark cluster manager) that has the power to request resources from Armada (e.g., to run additional pods) by submitting further Armada jobs. The operator job may do so either via the Armada job submission API or via a special set of Kubernets API endpoints that translate resource creation commands into Armada job submissions.
+
+In this way, Armada is in control of all resource creation and can manage cluster resources effectively.
+
+## Scheduler architecture
+
+At a high level, scheduling consists of:
+
+1. Global prioritisation
+2. Cluster assignment
+3. Node assignment
+
+The first two stages take place in the server and the third in the executor. We explain the first two stages here and the third stage in a later section. At any one time, each job is in one of the following three stages:
+
+1. Queued
+2. Pending
+3. Running
+
+Each job that is submitted is part of one of the `N` queues managed by the scheduler and is in the queued stage. Jobs within each queue are ordered by priority but there is no ordering between jobs in different queues. Jobs are moved from the queued stage into the pending stage one at a time and jobs in the pending stage are assigned to worker clusters on a first-in-first-out basis. Hence, all jobs in the pending stage are ordered and the global prioritisation stage consists of deciding in what order jobs should be moved from the queued to pending stage. Within each queue, jobs are moved from the queued to pending stage strictly in order of priority, i.e., for each queue, the next job to be moved into the pending stage is always the highest priority job in that queue.
```

Comment:
"There is no ordering between jobs in different queues". Surely we need to say something about relative queue priority?

Reply:
Actually you talk about it below, but I'd mention it here if only to say it's discussed in detail below.

Reply:
I've updated the text.